### ISSUE TITLE
Quick question regarding xform in dtiRawBuildBvecs.m

### ISSUE THREAD

**Issue by shrimpj** [vistalab/vistasoft#148]
Hi all,

I used freesurfer command mri_info to get xform information as I told by our radiographer. https://surfer.nmr.mgh.harvard.edu/fswiki/mri_info. He told me the only transformation he did was to flipped y axial when he converted raw DTI data to Nifti files. I assume this is the process of converting image from scanner coordinate to anatomical imaging coordinate.

I wonder which on should I use xforminfo or voxle to ras transformation matrix for this function? Because two of them are different.
   xform info: x_r =  -0.9974, y_r =  -0.0561, z_r =  -0.0462, c_r =    -0.5960
              
              : x_a =  -0.0578, y_a =   0.9977, z_a =   0.0364, c_a =    -9.5551
              
              : x_s =  -0.0441, y_s =  -0.0390, z_s =   0.9983, c_s =    10.5253

  voxel to ras transform:
                                 -1.7142  -0.0964  -0.2315   117.4822
               
                                 -0.0993   1.7147   0.1823  -114.6716
               
                                 -0.0757  -0.0670   4.9992   -27.8324
                
                                  0.0000   0.0000   0.0000     1.0000
Any suggestions would be grateful!

BW,
Joyce

**Comment by rfdougherty** [comment]
Hi Joyce,

Are you using a vendor-supplied diffusion sequence? If so, then the information to build the bvecs should be contained in the raw DICOM images, and good DICOM-to-NIFTI convertors (like dinifti or dcm2nii) should automatically generate the bvecs and bvals files for you. In that case, you would not need to use dtiRawBuildBvecs.

If you're using a custom pulse sequence that doesn't properly store the DWI gradient info in the images, then you will need more information from the sequence developer to get the gradient table and proper xfroms to generate bvecs and bvals.

bob

**Comment by shrimpj** [comment]
Hi Bob,

We used a fairly old sequence developed by Harvard with 26 directions. The original raw image we have are Idot files. Our technician converted it into DICOM images, then to NIFTI. The DTI directions information is stored in a dwepi.grads file. That is why I have to use ditRawBuildBvecs. Are you familiar with this type of raw images by any chance? Thank you for you reply!

BW,
Joyce

**Comment by rfdougherty** [comment]
No, I'm not familiar with idot files. You may just need to try different xforms to see what works. I'd start with no xform (i.e., eye(4)).

**Comment by shrimpj** [comment]
Thanks you so much for your reply! I tried all of them including xform=eye(3) as it 's given by the example. Why did you use eye(4)? I did not use the last column as I think it's irrelevant.

Joyce

**Comment by rfdougherty** [comment]
Yes, dtiRawBuildBvecs just needs a 3x3 rotation matrix. But if you give it a 4x4 affine transform, it will extract the 3x3 rotation matrix, so passing eye(4) is equivalent to eye(3).
